pymatgen/entries/tests/test_compatibility.py

```diff
@@ -72,8 +170,8 @@ def test_process_entry(self):
                                          'hash': '994537de5c4122b7f1b77fb604476db4'},
                                         {'titel': 'PAW_PBE O 08Apr2002',
                                          'hash': '7a25bc5b9a5393f46600a4939d357982'}]})
-        self.assertIsNone(self.compat.process_entry(entry))
-        self.assertIsNotNone(self.ggacompat.process_entry(entry))
+        self.assertIsNone(self.compat._process_entry(entry))
```

Comment:
Why are we testing private methods? Is it necessary for this to become private?

Reply:
It's not strictly necessary, but I think it's wise to make `process_entry` private going forward, for two reasons. First, we've made `process_entries` flexible enough to take in either a single `Entry` or a list. That way people don't have to change their scripts based on the number of entries they're processing. Second, some of the more sophisticated energy adjustment classes (including the new Aqueous) really have to analyze groups of entries at a time, it isn't really feasible to calculate the correction for just one entry.

I'm happy to remove the tests, but it seemed to me that more tests are always better than fewer, and I wanted them there to make sure we didn't break backward compatibility (although I suppose we already are since we've made `process_entry` private).

Reply:
Definitely don't remove the tests!

Reply:
I could add a public method to the `CorrectionsList` abstract class, so that all the legacy `Compatibility` classes will still have it available. But I'd really prefer `process_entry` not be a part of the base `Compatibility` interface.

Reply:
> First, we've made process_entries flexible enough to take in either a single Entry or a list. 

So should the tests not be equivalent to `.process_entries([entry])` ? And, in that case, should not `.process_entry` just do the same?

In general, not good form to have tests for private methods -- usually a sign the method should actually be public and documented.

> Second, some of the more sophisticated energy adjustment classes (including the new Aqueous) really have to analyze groups of entries at a time, it isn't really feasible to calculate the correction for just one entry.

Yes, makes sense.

Reply:
Yes, the tests are equivalent to `process_entries(entry)`, and that's exactly what `._process_entry` does "under the hood." Should I change the tests to use `process_entries`?

Reply:
Actually, the slight difference is that `process_entry` returns a single entry, whereas `process_entries` always returns a list.

Reply:
OK, I had forgotten that `.process_entry` was already a part of `CorrectionsList`. In that case I agree it makes sense to keep it public.

Reply:
Ok, I think it makes sense to put `process_entry()` on the same class as `process_entries()` , i.e. the abstract base class, and have it call `.process_entries([entry])` by default (if this needs to be overridden by a subclass that's fine, and is as it should be).

I hear you that it doesn't make sense in certain cases, might need to be overridden elsewhere, etc., but the alternatives (making it private, or not having it in the same class as `process_entries`) doesn't make much sense to me. That way the tests can proceed as previously.

Reply:
OK, done.